### PR TITLE
Editable sample requirements

### PR DESCRIPTION
**Pull request by alneberg**
A Vue setup of sample requirements that can be edited via genstat. A user with the role 'Sample requirements admin' can edit and publish new versions of the sample requirements table which is stored in statusdb. The code is in very large part copy pasted from the pricing setup and there might be code duplication that could be avoided with a bit more work.

Storing sample requirements in statusdb will make it possible to 
  - Automatically check user submitted sample sheets 
  - Highlight in genstat views whether a sample is violating any requirement
  - Simplify communication between Lab, Project coordinators and Users about sample requirements.
  - Flag samples for reason for failing reception control in a reception control report

And maybe more

### CODE REVIEW

File: run_dir/static/js/sample_requirements.js [hunk 1]
```diff
@@ -0,0 +1,1216 @@
...
+                }
+            }
+            this.sample_requirements[new_id] = new_sr
+            this.new_sample_requirements.add(new_id)
+            return new_id
+        },
+        newSampleRequirement() {
+            new_id = this.next_sample_requirement_id
+            /* Hacky way to be able to avoid typing all keys that a requirement should contain */
+            old_id = parseInt(new_id) - 1
```
[the remark this answers is not in the document]
Comment (alneberg) [reply run_dir/static/js/sample_requirements.js:110]: Yes, but it's meant to empty the values of it afterwards.

File: run_dir/static/js/sample_requirements.js [hunk 2]
```diff
@@ -0,0 +1,1216 @@
...
+                      <div class="modal fade" id="sample_requirements_view_changes_modal" tabindex="-1" role="dialog" aria-labelledby="sample_requirements_view_changes_modal_header">
+                        <div class="modal-dialog modal-lg" role="document">
+                          <div class="modal-content">
+                            <div class="modal-header">
+                              <h4 class="modal-title" id="sample_requirements_view_changes_modal_header">List of Changes and Validation Errors</h4>
+                              <button type="button" class="btn-close" data-dismiss="modal" aria-label="Close"></button>
+                            </div>
+                            <div class="modal-body">
+                              <template v-if="this.$root.no_validation_messages">
+                                <p>No validation erros found</p>
```
Comment (aanil) [inline run_dir/static/js/sample_requirements.js:893]: erros to errors?

File: run_dir/static/js/sample_requirements.js [hunk 3]
```diff
@@ -0,0 +1,1216 @@
...
+                <div class="modal-dialog" role="document">
+                  <div class="modal-content">
+                    <div class="modal-header">
+                      <h4 class="modal-title" id="publishDraftModalHeader">Publish new sample requirements</h4>
+                      <button type="button" class="btn-close" data-dismiss="modal" aria-label="Close"></button>
+                    </div>
+                    <div class="modal-body">
+                      <template v-if="this.$root.no_validation_messages">
+                        <p>Are you sure you want to publish the current draft?</p>
+                        <p>This will then become the default cost calculator used for all quotes.</p>
```
Comment (aanil) [inline run_dir/static/js/sample_requirements.js:599]: Should it be sample requirements rather  than cost calculator?

File: run_dir/static/js/sample_requirements.js (hunk 2 above)
Comment (alneberg) [reply run_dir/static/js/sample_requirements.js]: Yes, fixed.

File: run_dir/static/js/sample_requirements.js (hunk 3 above)
Comment (alneberg) [reply run_dir/static/js/sample_requirements.js]: Yes! Thank you! Fixed it now.

**Review by aanil — APPROVED** [review]
👍 Looks good to me!

**Pull request merged**